# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/thunderbird/hotfix-sparc.patch
# Copyright (C) 2023 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- firefox-118.0.2/xpcom/reflect/xptcall/md/unix/xptcinvoke_sparc_solaris.cpp.vanilla	2023-10-14 18:44:45.091217433 +0200
+++ firefox-118.0.2/xpcom/reflect/xptcall/md/unix/xptcinvoke_sparc_solaris.cpp	2023-10-14 18:57:05.548236848 +0200
@@ -22,7 +22,7 @@
     uint32_t result = 0;
     for(uint32_t i = 0; i < paramCount; i++, s++)
     {
-        if(s->IsPtrData())
+        if(s->IsIndirect())
         {
             result++;
             continue;
@@ -90,15 +90,7 @@
     for(uint32_t i = 0; i < l_paramCount; i++, l_d++, l_s++)
     {
 	if (regCount < 5) regCount++;
-        if(l_s->IsPtrData())
-        {
-            if(l_s->type == nsXPTType::T_JSVAL)
-            {
-              // On SPARC, we need to pass a pointer to HandleValue
-              *((void**)l_d) = &l_s->ptr;
-            } else
-            {
-              *((void**)l_d) = l_s->ptr;
-            }
+        if (l_s->IsIndirect()) {
+            *((void**)l_d) = &l_s->val;
             continue;
         }
